nova/virt/vmwareapi/vmops.py

```diff
@@ -2181,22 +2169,38 @@ def is_instance_in_resource_pool(self, instance):
             LOG.debug("Failed to find instance", instance=instance)
             return False
 
-    def _get_instance_props(self, vm_ref):
-        lst_properties = ["config.instanceUuid",
-                          "runtime.powerState",
-                          "summary.guest.toolsStatus",
-                          "summary.guest.toolsRunningStatus",
-                         ]
+    def _get_instance_props(self, instance, lst_properties=None,
+                            skip_update=False):
+        lst_properties = (lst_properties or
+            ["config.instanceUuid",
+             "runtime.powerState",
+             "summary.guest.toolsStatus",
+             "summary.guest.toolsRunningStatus"])
 
-        self.update_cached_instances()
+        if not skip_update:
+            self.update_cached_instances()
+        vm_ref = vm_util.get_vm_ref(self._session, instance)
         vm_props = vm_util._VM_VALUE_CACHE.get(vm_ref.value, {})
 
         if set(vm_props.keys()).issuperset(lst_properties):
             return vm_props
-        else:
-            return self._session._call_method(
-                vutil, "get_object_properties_dict",
-                vm_ref, lst_properties)
+
+        try:
+            return self._session._call_method(vutil,
+                                            "get_object_properties_dict",
+                                            vm_ref, lst_properties)
+        except vexc.ManagedObjectNotFoundException:
+            raise exception.InstanceNotFound(instance_id=instance.uuid)
+
+    def _get_instance_property(self, instance, prop):
+        vm_ref = vm_util.get_vm_ref(self._session, instance)
+        try:
+            return self._session._call_method(vutil,
+                                              "get_object_property",
+                                              vm_ref,
+                                              prop)
+        except vexc.ManagedObjectNotFoundException:
```

Comment:
We have the following helper function, but we seem to not use it here. Is there any reason for that?
```
nova/virt/vmwareapi/vim_util.py:def get_object_property(session, mo_ref, property):
```

I'm wondering if we need that helper function at all, it just shortens the line basically by hiding the `_call_method` part and the `"get_object_property"`.

Reply:
I can't remember a good reason, I'll change it.